src/main/java/hudson/maven/MavenEmbedderUtils.java

```diff
@@ -205,34 +212,39 @@ public static MavenInformation getMavenVersion(File mavenHome) throws MavenEmbed
                         + "'. Are you sure that this is a valid maven home?");
             }
             URLConnection uc = resource.openConnection();
-            if (uc instanceof JarURLConnection) {
-                final JarURLConnection connection = (JarURLConnection)uc;
-                final String entryName = connection.getEntryName();
-                try {
-                    jarFile = connection.getJarFile();
-                    final JarEntry entry = (entryName != null && jarFile != null) ? jarFile.getJarEntry(entryName) : null;
-                    if (entry != null) {
-                        inputStream = jarFile.getInputStream(entry);
-                        Properties properties = new Properties();
-                        properties.load( inputStream );
-                        information = new MavenInformation( properties.getProperty( "version" ) , resource.toExternalForm() );
-                    }
-                } finally {
-                    if (jarFile != null) {
-                        jarFile.close();
-                    }
+            uc.setUseCaches(false);
+            InputStream istream = uc.getInputStream();
+            try {
+                if (preopertiesPreloadHook != null) {
+                    preopertiesPreloadHook.call();
                 }
-            }
+                Properties properties = new Properties();
+                properties.load( istream );
+                information = new MavenInformation( properties.getProperty( "version" ) , resource.toExternalForm() ); 
+            } finally {
+                istream.close();
+            } 
         } catch ( IOException e ) {
             throw new MavenEmbedderException( e.getMessage(), e );
         } finally {
-            IOUtil.close( inputStream );
             Thread.currentThread().setContextClassLoader( original );
+            closeIt(realm);
         }
 
         return information;
     }
     
+    private static final void closeIt(@CheckForNull Closeable obj) throws MavenEmbedderException {
+        if (obj == null) {
+            return;
+        }
+        try {
+            obj.close();
+        } catch(IOException ex) {
+            throw new MavenEmbedderException("Failed to close " + obj, ex);
```

Comment:
I am fine with that. Should be another PR OTOH, because it will also require changes in dependent plugins